packages/@aws-cdk/aws-secretsmanager/lib/secret.ts

```diff
@@ -233,6 +252,29 @@ export class Secret extends SecretBase {
     return Secret.fromSecretAttributes(scope, id, { secretArn });
   }
 
+  /**
+   * Imports a secret by secret name; the ARN of the Secret will be set to the secret name.
+   * A secret with this name must exist in the same account & region.
+   */
+  public static fromSecretName(scope: Construct, id: string, secretName: string): ISecret {
+    return new class extends SecretBase {
+      public readonly encryptionKey = undefined;
+      public readonly secretArn = secretName;
+      public readonly secretName = secretName;
+      protected readonly autoCreatePolicy = false;
+      // Overrides the secretArn for grant* methods, where the secretArn must be in ARN format.
+      // Also adds a wildcard to the resource name to support the SecretsManager-provided suffix.
+      protected arnForPolicies() {
+        return Stack.of(this).formatArn({
+          service: 'secretsmanager',
+          resource: 'secret',
+          resourceName: this.secretName + '*',
```

Comment:
`+ '-*'`? To make sure the only part we wildcard is the generated suffix?

Reply:
I considered that, and even coded it up that way initially, then changed my mind. This impacts your other comment as well. Happy for a second opinion here.

If the user provides only the "friendly" version of the secret name (e.g., `My-Special-Secret`), then yes, we should prepend with `'-*'` to do the right thing. However, if a user provides the whole name (e.g., `My-Special-Secret-123bdc`), then adding `-*` will break the permissions; this is fairly subtle breakage, so I could see it being hard to diagnose. The gotcha is that we can't tell if the user's provided the full secret name or just the "friendly" name. 

Also, using `-*` doesn't guarantee we'll only wildcard the suffix. If I have a secret `My-Special-Secret` that I want access to, I can create another secret `My-Special`, and `grantRead` on that will give me access to `My-Special-Secret` too. This is admittedly an edge case with secret names that include `-`, but it's possible.

So in the end I weighed the benefits (slightly better, but far from foolproof security) and risks (really subtle breakage if the user supplies the "full" secret name) and opted for this. Totally willing to have my mind changed here though.